src/main/scala/codacy/plugins/test/ITest.scala

```diff
@@ -3,59 +3,143 @@ package codacy.plugins.test
 import java.io.File
 import java.nio.file.Path
 
-import codacy.plugins.docker.{DockerPlugin, Pattern, Result}
 import codacy.utils.Printer
+import com.codacy.analysis.core
+import com.codacy.analysis.core.model.{FileError, Issue, Pattern, ToolResult}
+import com.codacy.plugins.api._
+import com.codacy.plugins.api.languages.{Language, Languages}
+import com.codacy.plugins.results.traits.DockerTool
+import com.codacy.plugins.utils.PluginHelper
+
+import scala.util.{Failure, Success, Try}
+
+final case class DockerImage(name: String, version: String) {
+  override def toString: String = {
+    s"$name:$version"
+  }
+}
 
 trait ITest {
   val opt: String
 
-  def run(plugin: DockerPlugin, testSources: Seq[Path], dockerImageName: String, optArgs: Seq[String]): Boolean
+  def run(testSources: Seq[Path], dockerImage: DockerImage, optArgs: Seq[String]): Boolean
 
-  protected def filterResults(sourcePath: Path, files: Seq[File], patterns: Seq[Pattern], results: Seq[Result]): Seq[Result] = {
-    val receivedResultsTotal = results.length
 
-    if (results.nonEmpty) {
-      Printer.green(s"$receivedResultsTotal results received.")
-    } else {
-      Printer.red("No results received!")
+  protected def findLanguages(testSources: Seq[Path], dockerImage: DockerImage): Set[Language] = {
+    sys.props.get("codacy.tests.languages").map(_.split(",").flatMap(Languages.fromName).to[Set])
+      .orElse {
+        (core.tools.Tool.availableTools ++ PluginHelper.dockerEnterprisePlugins).collectFirst {
+          case tool if tool.dockerName == dockerImage.name && tool.dockerTag == dockerImage.version =>
+            tool.languages
+        }
+      }.getOrElse {
+      (for {
+        sourcePath <- testSources
+        testFile <- new TestFilesParser(sourcePath.toFile).getTestFiles
+        language <- Languages.fromName(testFile.language.toString)
+      } yield language) (collection.breakOut)
     }
+  }
 
-    val relativeFiles = files.map(file => sourcePath.relativize(file.getAbsoluteFile.toPath).toString)
-    val (filteredFileResults, otherFilesResults) = results.partition { result =>
-      relativeFiles.contains(result.filename)
+  protected def createDockerTool(languages: Set[Language], dockerImage: DockerImage): DockerTool = {
+    val dockerImageName = dockerImage.name
+    val dockerImageVersion = dockerImage.version
+
+    new DockerTool(dockerName = dockerImageName, true, languages,
+      dockerImageName, dockerImageName, dockerImageName, "", "", "", needsCompilation = false,
+      needsPatternsToRun = true, hasUIConfiguration = true) {
```

Comment:
Having everything as a named argument would help understanding the passed values.

Reply:
👍  done